lib/chef/resource/locale.rb

```diff
@@ -25,67 +25,115 @@ class Locale < Chef::Resource
+               description: "Sets the default system language.",
+               regex: [LOCALE_REGEX],
+               validation_message: "The provided lang not valid. It should be a non-empty string without any leading whitespaces."
+
+      property :lc_env, Hash,
+               description: "A Hash of LC_* env variables in the form of ({'LC_ENV_VARIABLE' => 'VALUE'}).",
+               default: lazy { {} },
+               coerce: proc { |h|
+                         if h.respond_to?(:keys)
+                           invalid_keys = h.keys - LC_VARIABLES
+                           unless invalid_keys.empty?
+                             error_msg = "Key of option lc_env must be equal to one of: \"#{LC_VARIABLES.join('", "')}\"!  You passed \"#{invalid_keys.join(', ')}\"."
+                             raise Chef::Exceptions::ValidationFailed, error_msg
+                           end
+                         end
+                         unless h.values.all? { |x| x =~ LOCALE_REGEX }
+                           error_msg = "Values of option lc_env should be non-empty string without any leading whitespaces."
+                           raise Chef::Exceptions::ValidationFailed, error_msg
+                         end
+                         h
+                       }
 
-      property :lc_all, String,
-               default: "en_US.utf8",
-               description: "Sets the fallback system language."
+      # @deprecated Use {#lc_env} instead of this property.
+      #   {#lc_env} uses Hash with specific LC var as key.
+      # @raise [Chef::Deprecated]
+      #
+      def lc_all(arg = nil)
+        unless arg.nil?
+          Chef.deprecated(:locale_lc_all, "Changing LC_ALL can break Chef's parsing of command output in unexpected ways.\n Use one of the more specific LC_ properties as needed.")
+        end
+      end
 
       action :update do
         description "Update the system's locale."
-
-        if node["init_package"] == "systemd"
-          # on systemd settings LC_ALL is (correctly) reserved only for testing and cannot be set globally
-          execute "localectl set-locale LANG=#{new_resource.lang}" do
-            # RHEL uses /etc/locale.conf
-            not_if { up_to_date?("/etc/locale.conf", new_resource.lang) } if ::File.exist?("/etc/locale.conf")
-            # Ubuntu 16.04 still uses /etc/default/locale
-            not_if { up_to_date?("/etc/default/locale", new_resource.lang) } if ::File.exist?("/etc/default/locale")
+        begin
+          unless up_to_date?
+            converge_by "Updating System Locale" do
+              generate_locales unless unavailable_locales.empty?
+              update_locale
+            end
           end
-        elsif ::File.exist?("/etc/sysconfig/i18n")
-          locale_file_path = "/etc/sysconfig/i18n"
+        rescue
+          # It might affect debugging
+          raise "#{node['platform']} platform not supported by the chef locale resource."
```

Comment:
Add "If you believe this is in error please file an issue at https://github.com/chef/chef/issues"

Reply:
Thanks. Added.